src/megatron/energon/flavors/base_dataset.py

```diff
@@ -122,6 +122,60 @@ class Sample(ABC, PinMemoryMixin, ExtendableDataclassMixin):
     #: A dataset may define a subflavors to distinguish between samples of the same sample type.
     __subflavors__: Optional[Dict[str, Any]]
 
+    def to_joined(self, type: Type[T_sample], **kwargs) -> T_sample:
```

Comment:
Not sure what exactly the last one means, since you already have a method `from_joined` but I think you mean something else here. Also it has nothing to do with "join", right?

Maybe `TextSample.derive_from(other_sample, key1=val1, key2=val2)`, which would copy the basic keys from `other_sample`?

Reply:
Okay, agreed, does not have to be `from_joined`. Yes, that's what I mean.

Reply:
Ok, now made it as proposed: `TextSample.derive_from`